Formula/h/hdf5.rb

```diff
@@ -1,10 +1,10 @@
 class Hdf5 < Formula
   desc "File format designed to store large amounts of data"
   homepage "https://www.hdfgroup.org/HDF5"
-  url "https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.14/hdf5-1.14.3/src/hdf5-1.14.3.tar.bz2"
-  sha256 "9425f224ed75d1280bb46d6f26923dd938f9040e7eaebf57e66ec7357c08f917"
+  url "https://github.com/HDFGroup/hdf5/releases/download/hdf5_1.14.4.2/hdf5-1.14.4-2.tar.gz"
+  version "1.14.4"
```

Comment:
1.14.4-3 is now out https://github.com/HDFGroup/hdf5/tree/hdf5_1.14.4.3

May need to track the extra number in version:

> * The release tag in GitHub will be hdf5_1.14.4.3
> 
>   We used to use tags with names like hdf5-1_14_3, but we've been informed
>   that dashes are forbidden in version strings by many package management
>   schemes. If a patch release is necessary, we'll simply append another
>   dot and number - e.g. hdf5_1.14.4.3 (instead of our previous hdf5-1_14_4-3).

Reply:
Done